plugins/modules/supported.py

```diff
@@ -0,0 +1,360 @@
+            products = []
+            for param_name, version in self.product_versions.items():
+                product_name = self.PRODUCT_NAME_MAPPING[param_name]
+                # URL encode spaces in both product name and version
+                # Use urllib.parse.quote for proper URL encoding
+                product_version_string = f"{product_name}-{version}"
+                encoded_product_version = quote(product_version_string)
+                products.append(encoded_product_version)
+                self.filters_applied[param_name] = version
+
+            if products:
+                # Join products with commas for a single PRODUCT parameter
+                product_condition = f"PRODUCT={','.join(products)}"
+                conditions.append(product_condition)
+
+        return ";".join(conditions)
+
+    def _build_api_url(self):
+        """
+        Build the complete API URL with query parameters.
+
+        Returns:
+            str: Complete API URL
+        """
+        conditions = self._build_query_conditions()
+
+        if conditions:
+            # Add trailing semicolon as shown in the curl example
+            # Don't URL encode the entire condition string, only spaces are encoded
+            self.api_url = f"{self.BASE_URL}?condition={conditions};"
+        else:
+            self.api_url = self.BASE_URL
+
+        return self.api_url
+
+    def process(self):
+        """
+        Fetch support matrix data from the Cloudera API using Ansible's fetch_url.
+
+        Returns:
+            bool: True if successful, False otherwise
+        """
+
+        try:
+            # Build the API URL
+            api_url = self._build_api_url()
+
+            # Prepare headers
+            headers = {
+                "Accept": "*/*",
+                "Content-Type": "application/json",
+                "User-Agent": "Ansible/cloudera.cluster",
+            }
+
+            # Use Ansible's fetch_url
+            response, info = fetch_url(
+                self.module,
+                api_url,
+                headers=headers,
+                timeout=self.timeout,
+            )
+
+            if info.get("status") != 200:
+                self.module.fail_json(
+                    msg=f"HTTP error occurred: {info.get('msg', 'Unknown error')}",
+                    api_url=api_url,
+                    http_status=info.get("status"),
+                    http_reason=info.get("msg"),
+                )
+
+            response_text = to_text(response.read()) if response else ""
+
+            # Parse JSON response
+            if response_text:
+                try:
+                    self.support_matrix_data = json.loads(response_text)
+                except json.JSONDecodeError as e:
+                    self.module.fail_json(
+                        msg=f"Failed to parse JSON response: {to_native(e)}",
+                        api_url=api_url,
+                        response_text=response_text[

```

Comment:
I believe this is taken care of by the to_native() call, however, you might want to check that, as to_native could just return the error message and not the full error object.